ConsistencyManagerTests/ProjectionTests.swift

```diff
@@ -0,0 +1,168 @@
+     We're going to update this with a new model which has two children with the same id (but different projections).
+     We will verify that the original child get's both updates.
+     */
+    func testUpdatingModelWithTwoProjections() {
+        let child = ProjectionTreeModel(type: .both, id: 1, data: 2, otherData: 4, child: nil, otherChild: nil)
+        let testModel = ProjectionTreeModel(type: .both, id: 0, data: 1, otherData: 1, child: child, otherChild: nil)
+
+        let consistencyManager = ConsistencyManager()
+        let listener = TestListener(model: testModel)
+
+        addListener(listener, toConsistencyManager: consistencyManager)
+
+        listener.contextClosure = { context in
+            XCTAssertEqual(context as? String, "context")
+        }
+
+        listener.updateClosure = { _, updates in
+            XCTAssertEqual(updates.changedModelIds.count, 2)
+            XCTAssertEqual(updates.deletedModelIds.count, 0)
+            XCTAssertTrue(updates.changedModelIds.contains("1"))
+            XCTAssertTrue(updates.changedModelIds.contains("0"))
+        }
+
+        // This should cause both children to update one of their fields
+        let updateChild = ProjectionTreeModel(type: .data, id: 1, data: 3, otherData: nil, child: nil, otherChild: nil)
+        let updateOtherChild = ProjectionTreeModel(type: .otherData, id: 1, data: nil, otherData: 3, child: nil, otherChild: nil)
+        let updateModel = ProjectionTreeModel(type: .both, id: 2, data: 5, otherData: 5, child: updateChild, otherChild: updateOtherChild)
+
+        updateWithNewModel(updateModel, consistencyManager: consistencyManager, context: "context")
+
+        // Should have updated the model of the listener
+        if let testModel = listener.model as? ProjectionTreeModel {
+            // Both fields should have been updated. One from each child model.
+            XCTAssertEqual(testModel.child?.data, 3)
+            XCTAssertEqual(testModel.child?.otherData, 3)
+        } else {
+            XCTFail()
+        }
+    }
+
+    /**
+     This tests a tree with one child (.data projection).
+     We're going to update this with a .otherData projection.
+     Since it doesn't care about any of these changes (nothing has actually changed), this should short curcuit and not call update.
+     */
+    func testNoUpdateWhenNothingChanges() {
+        let child = ProjectionTreeModel(type: .data, id: 1, data: 2, otherData: nil, child: nil, otherChild: nil)
+        let testModel = ProjectionTreeModel(type: .both, id: 0, data: 1, otherData: 1, child: child, otherChild: nil)
+
+        let consistencyManager = ConsistencyManager()
+        let listener = TestListener(model: testModel)
+
+        addListener(listener, toConsistencyManager: consistencyManager)
+
+        listener.updateClosure = { _, _ in
+            XCTFail()
+        }
+
+        // This should cause both children to update one of their fields
```

Comment:
No update occurs here, because the children with id=1 are of different projection types that do not overlap